web/src/main/java/org/akaza/openclinica/service/ParticipantServiceImpl.java

```diff
@@ -377,8 +345,188 @@ public int getSubjectCount(StudyBean currentStudy) {
+                studyParticipantDTO.setSubjectKey(subject.getStudySubjectId());
+
+                ParticipantValidator participantValidator = new ParticipantValidator(dataSource);
+                participantValidator.setBulkMode(true);
+                Errors errors = null;
+
+                DataBinder dataBinder = new DataBinder(subject);
+                errors = dataBinder.getBindingResult();
+                participantValidator.validateBulk(subject, errors);
+
+                if (errors.hasErrors()) {
+                    ArrayList validerrors = new ArrayList(errors.getAllErrors());
+                    Iterator errorIt = validerrors.iterator();
+                    while (errorIt.hasNext()) {
+                        ObjectError oe = (ObjectError) errorIt.next();
+
+                        if (oe.getCode() != null) {
+                            errorMsgs.add(oe.getCode());
+                        } else {
+                            errorMsgs.add(oe.getDefaultMessage());
+                        }
+                    }
+                }
+
+                if (errorMsgs != null && errorMsgs.size() != 0) {
+                    StudySubjectBean byLabelAndStudy = getStudySubjectDao().findByLabelAndStudy(subject.getPersonId(), study);
+
+                    OCParticipateImportDTO p = new OCParticipateImportDTO();
+                    p.setRow(index);
+                    p.setParticipantId(subject.getStudySubjectId());
+
+                    if (byLabelAndStudy != null) {
+                        p.setParticipantOid(byLabelAndStudy.getOid());
+                        p.setParticipateStatus(byLabelAndStudy.getStatus().getName());
+                    }
+                    p.setStatus("Failed");
+
+                    p.setMessage(errorMsgs.get(0));
+                    ocParticipateImportDTOs.add(p);
+                } else {
+                    String label = null;
+                    try {
+                        label = createParticipant(subject, study, accessToken, customerUuid, user, locale);
+                    } catch (Exception e) {
+                        logger.error("Error in createParticipant:" + e);
+                    }
+                    StudySubjectBean subjectBean = this.getStudySubjectDAO().findByLabel(label);
+                    studyParticipantDTO.setSubjectOid(subjectBean.getOid());
+
+                    OCParticipateImportDTO p = new OCParticipateImportDTO();
+                    p.setRow(index);
+                    p.setParticipantId(subjectBean.getLabel());
+                    p.setParticipantOid(subjectBean.getOid());
+                    p.setParticipateStatus(subjectBean.getStatus().getName());
+                    p.setStatus("Success");
+                    p.setMessage("");
+                    ocParticipateImportDTOs.add(p);
 
+                }
+                ++index;
+            }
+            writeToFile(ocParticipateImportDTOs, studyOid, fileName);
```

Comment:
Should we be writing to file after each participant is added such that if an uncaught exception is thrown while processing a participant the logs for previous entries are already written to file?